docs/proposals/29-autoscaling-storage-volumes.md

```diff
@@ -0,0 +1,335 @@
+infrastructure supports it.
+ 
+## Motivation
+Logs are vital for understanding and troubleshooting Gardener clusters. Yet we operate under tight resource constraints
+and logging stack PVs default to a static size of 30GB, for ALL shoots. To prevent reaching this limit and bringing down
+the logging stack, there is a curator sidecar which periodically removes older logs. As a result, in some very large or
+busy clusters, this strategy allows retention of logs for only for 2-3 days.
+
+On the other hand, while some clusters experience storage starvation, others operate with just 5-10% of their observability
+volumes utilised.
+
+The present initiative is focused on observability storage volumes, and aims to:
+- Optimise volume utilisation
+- Ensure that high-throughput clusters receive sufficient observability storage space
+
+The goals are achieved by the following means:
+- A Gardener controller is introduced to auto-scale PVs (expand only)
+- Observability volumes are selectively marked for action by said controller
+- Initial volume size is reduced
+- Where the storage infrastructure does not support volume resizing, the preexisting, static-size volume sizing strategy is employed
+
+The current initiative is limited to scaling observability components - _Vali_ and _Prometheus_.
+However, the long term vision is, once the PV autoscaling implementation matures enough, to also use it to offer PV
+auto-scaling for shoot (i.e. customer) volumes.
+
+There are no plans to extend the present initiative to shoot Etcd volumes. Instead, a more advanced, Etcd-specific
+storage management functionality is envisioned in `etcd-druid`. For details, see
+[`etcd-druid` feature request issue#481](https://github.com/gardener/etcd-druid/issues/481#issuecomment-1346402013).
+
+### Goals
+- General ability to autoscale seed PVs.
+  - Ability is conditional on PVC's storage class supporting resizing.
+  - Autoscaling is inactive by default.
+  - Ability to selectively activate autoscaling per PVC.
+  - Ability to trigger scaling based on relative volume utilisation ratio (`<used bytes> / <capacity bytes>`). Trigger
+    threshold configurable via code.
+  - Ability to configure absolute upper scaling limit per PVC.
+  - Autoscaling causes minimal interruption to scaled workloads.
+- Autoscaling ability activated for observability volumes (`prometheus` and `vali`) in seeds'
+  `garden` namespace, and in shoot namespaces.
+- Absolute upper scaling limit configured on each observability volume.
+- Reduce the initial size of a volume, when autoscaling is activated for that volume, if such reduction is aligned with
+  general Gardener goals (cost, reliability).
+- Some metrics, which represent an overview of the PVC autoscaling state per seed, available through one or more of the seed
+  system's Prometheus instances.
+- Provide operational documentation for Gardener Operators.
+- Ability to enable/disable the feature per seed, via feature gate.
```

Comment:
Also, what happens if we create a cluster while autoscaling is enabled, i.e. with a small pvc size, and autoscaling is disabled at a later point in time. Will the pvc increased to the "default" or will it stay "small"?

Reply:
@ScheererJ Yes. After the recent update to the GEP, autoscaling is enabled on preexisting volumes

@plkokanov Yes, after the following sequence of events:
1. Autoscaling feature enabled
2. Autoscaling enabled on volume X
3. Autoscaler expands volume X beyond the static level of capacity, which would be allocated if the feature was  disabled
4. Autoscaling feature disabled
...the volume would be deleted and redeployed with capacity at the static level.

@ScheererJ It will be expanded to "default" capacity.
If the autoscaling feature is disabled, a volume with a small size will be resized by the code @plkokanov mentions above. If the storage class still supports resize at that time, it will be expanded to the "default" static capacity.